rfcs/20190802-model-garden-redesign.md

```diff
@@ -0,0 +1,207 @@
+accuracy tests for TensorFlow on different types of accelerators. All official
+TF2.0 models are required to provide accuracy tests and these tests will be
+automatically expanded to performance tests for continuous regression testing
+and monitoring.
+
+## Model Garden Sustainability
+
+### Model Launch Criteria
+To ensure that official models are well-maintained and tested, we are going to enforce the following criteria for launching a new model in the official model garden, except for staging folder:
+
+*   Follow the best practice guideline for each model category.
+*   Unit tests to verify the basics of the model.
+*   Integrate the model to benchmark testing to ensure model’s accuracy should be on par with the original paper / SOTA results.
+*   README with commands and procedures to reproduce the SOTA results, including:
+    *   Input data generation if necessary
+    *   Model execution, including all hyperparameters.
+
+### Community contribution and staging
+
+Due to fast ML development, we can’t possibly support all best-in-class models
+up to date on our own. We highly encourage users to contribute to the official
+model garden. After model garden refactoring (Phase 1), we plan to provide
+a full list of wanted models to tensorflow community and encourage tensorflow
+users to claim and contribute the models to the model garden.
+
+We have different requirements from unifying interface, supporting all the chips
+and platforms and enabling benchmarks for reference models. Thus, we could have
+different stages of models. As we may have immediate needs to add some quick
+models for benchmark and debugging, we will provide a staging folder to host
+some drafts of SOTA or popular models. Once the staging models can converge and
+support major functionalities of standard official models, we can judge whether
+they meet the launch standard and migrate to official models or migrate them to
+benchmark references.
+
+### Maintenance and Deprecation
+
+Given the nature of this repository, old models may become less and less
+useful to the community as time goes on. In order to keep the repository
+sustainable, we will be performing bi-annual reviews of our models to ensure
+everything still belongs to the repo. For models to be retired, the current plan
+is to move them to the archive directory and these models won't run regression
+tests to ensure the quality and convergence.
+
+The following details the policy for models in mature and staging phases:
+
+*   Models graduated from staging subdirectory
+
+    The models will be maintained by the model garden team. After we start to
+    accept community contributions, we will put the contributors as model owners.
+
+    These models will have continuous convergence and performance testing to
+    make sure no regression. In general, we won’t deprecate these models unless:
+    *   the model isn’t compatible with the TF APIs any more and have to be replaced by a new version
```

Comment:
Does this imply that the model will only target *latest* version of TF?

Reply:
We plan to do model garden release for major TF release, e.g. TF 2.1, 2.2. Thanks!

Reply:
Cool! Thanks for your reply.